Source/Core/CustomProjection.js

```diff
@@ -0,0 +1,311 @@
+define([
+        './Cartesian3',
+        './Cartographic',
+        './Check',
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './DeveloperError',
+        './Ellipsoid',
+        './getAbsoluteUri',
+        './isDataUri',
+        './loadAndExecuteScript',
+        '../ThirdParty/when'
+    ], function(
+        Cartesian3,
+        Cartographic,
+        Check,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Ellipsoid,
+        getAbsoluteUri,
+        isDataUri,
+        loadAndExecuteScript,
+        when) {
+    'use strict';
+
+    /**
+     * {@link MapProjection} that uses custom project and unproject functions defined in an external file.
+     *
+     * The external file must contain a function named <code>createProjectionFunctions</code> that implements the
+     * <code>CustomProjection~factory</code> interface to provide <code>CustomProjection~project</code> and
+     * <code>CustomProjection~unproject</code> functions to a callback.
+     *
+     * Scenes using CustomProjection will default to <code>MapMode2D.ROTATE</code> instead of <code>MapMode2D.INFINITE_SCROLL</code>.
+     *
+     * @alias CustomProjection
+     * @constructor
+     *
+     * @param {String} url The url of the external file.
```

Comment:
`external file` is misleading terminology here, if this is a data uri then there's no file involved (also files doesn't necessarily exist on the internet, this is just an endpoint that returns the JavaScript containing the values).  Remove all `external file` terminology and come up with a more general way to describe it.